projects/giflib/egif_fuzz_common.cc

```diff
@@ -1,26 +1,36 @@
 #include "egif_fuzz_common.h"
-#define GIF_IMAGE_WIDTH  100
+#define GIF_IMAGE_WIDTH 100
 // This is rgb byte stream length per horizontal line = GIF_IMAGE_WIDTH * 3
-#define GIF_IMAGE_LINE   300
+#define GIF_IMAGE_LINE 300
 
 extern "C" void PrintGifError(int ErrorCode);
 
-int stub_output_writer(GifFileType* gifFileType, const uint8_t* buf, int len)
+int stub_output_writer(GifFileType *gifFileType, const uint8_t *buf, int len)
 {
-	struct gifUserData* gud = (struct gifUserData*) gifFileType->UserData;
+	struct gifUserData *gud = (struct gifUserData *)gifFileType->UserData;
 
 	if (gud == NULL || gud->gifData == NULL || len == 0)
 		return 0;
-
-	memcpy(gud->gifData, buf, len);
-	gud->gifData += len;
+	if (gud->allocatedSize < (gud->gifLen + len))
+	{
+		// Reallocate gifFileType
+		int newSize = gud->allocatedSize * 2;
+		uint8_t *oldGud = gud->gifData;
+		gud->gifData = (uint8_t *)realloc(oldGud, newSize);
+		if (gud->gifData == NULL)
+		{
+			gud->gifData = oldGud;
+			return 0;
+		}
+		gud->allocatedSize = newSize;
+	}
+	memcpy(gud->gifData + gud->gifLen, buf, len);
 	gud->gifLen += len;
 	return len;
 }
 
 // RGB to GIF converter
-static
-bool rgb_to_gif(const uint8_t* data, size_t size)
+static bool rgb_to_gif(const uint8_t *data, size_t size)
```

Comment:
Ignoring the formatting changes since I feel they were due to an IDE.

Reply:
Yup - sorry for polluting the commit.

Reply:
No worries, just wanted to mention I didn't review these lines ;)